<launch>
  <!-- RRT* 3D Path Planner Node -->
  <node name="rrt_star_planner" pkg="rrt_star_planner" type="rrt_star_planner_node" output="screen">
    <!-- RRT* Parameters -->
    <param name="max_step_size" value="0.5" />
    <param name="goal_tolerance" value="0.3" />
    <param name="robot_radius" value="0.5" />
    <param name="max_iterations" value="5000" />
    <param name="rewire_radius" value="1.0" />
    <param name="goal_bias_probability" value="0.1" />
    
    <!-- Sampling bounds -->
    <param name="x_min" value="-10.0" />
    <param name="x_max" value="10.0" />
    <param name="y_min" value="-10.0" />
    <param name="y_max" value="10.0" />
    <param name="z_min" value="0.0" />
    <param name="z_max" value="5.0" />
    
    <!-- Topic remapping if needed -->
    <remap from="robot_pose" to="/robot/pose" />
    <remap from="obstacle_pointcloud" to="/mapping/pointcloud" />
  </node>
</launch>